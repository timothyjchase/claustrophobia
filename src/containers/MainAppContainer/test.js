import React from 'react'
import { MainAppContainer, mapStateToProps } from './index'

describe('<MainAppContainer />', () => {
  describe('with scenarioKey = TEST_SCENARIO', () => {
    const props = {
      currentState: {
        scenarioKey: 'TEST_SCENARIO',
      },
      startGame: jest.fn(),
      resetGame: jest.fn(),
      undoLastChange: jest.fn(),
    }
    const component = shallow(<MainAppContainer {...props} />)
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenarioKey = null', () => {
    const props = {
      currentState: {
        scenarioKey: null,
      },
      startGame: jest.fn(),
      resetGame: jest.fn(),
      undoLastChange: jest.fn(),
    }
    const component = shallow(<MainAppContainer {...props} />)
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})

describe('mapStateToProps', () => {
  describe('with basic state', () => {
    const state = { current: {} }
    it('should return currentState', () => {
      expect(mapStateToProps(state)).toEqual({ currentState: {} })
    })
  })
})
