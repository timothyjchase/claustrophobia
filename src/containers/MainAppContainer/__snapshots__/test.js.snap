// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainAppContainer /> with scenarioKey = TEST_SCENARIO should match snapshot 1`] = `
<MainApp
  currentState={
    Object {
      "scenarioKey": "TEST_SCENARIO",
    }
  }
  resetGame={[MockFunction]}
  scenarioName="Test Scenario"
  startGame={[MockFunction]}
  undoLastChange={[MockFunction]}
/>
`;

exports[`<MainAppContainer /> with scenarioKey = null should match snapshot 1`] = `
<MainApp
  currentState={
    Object {
      "scenarioKey": null,
    }
  }
  resetGame={[MockFunction]}
  startGame={[MockFunction]}
  undoLastChange={[MockFunction]}
/>
`;
